contracts/misc/NFTFloorOracle.sol

```diff
@@ -2,452 +2,160 @@
-        PriceInformation memory _priceInfo,
-        uint256 _twap
-    ) internal returns (bool, uint256) {
-        uint256 currentBlock = block.number;
-        uint256 currentTwap = _priceInfo.twap;
-
-        _feederRegistrar.feederPrice[msg.sender].twap = _twap;
-        _feederRegistrar.feederPrice[msg.sender].updatedAt = currentBlock;
-
-        //first time just use the feeding value
-        if (currentTwap == 0) {
-            return (true, _twap);
-        }
-        //use memory here so allocate with maximum length
-        address[] memory _feeders = feeders;
-        uint256[] memory validPriceList = new uint256[](_feeders.length);
-        uint256 validNum = 0;
-        //aggeregate with price from all feeders
-        for (uint256 i = 0; i < _feeders.length; i++) {
-            PriceInformation memory priceInfo = _feederRegistrar.feederPrice[
-                _feeders[i]
-            ];
-            uint256 diffBlock = currentBlock - priceInfo.updatedAt;
-            if (
-                priceInfo.updatedAt > 0 && diffBlock <= _config.expirationPeriod
-            ) {
-                validPriceList[validNum] = priceInfo.twap;
-                validNum++;
-            }
-        }
-        if (validNum < MIN_ORACLES_NUM) {
-            return (false, currentTwap);
-        }
-        _quickSort(validPriceList, 0, int256(validNum - 1));
-        return (true, validPriceList[validNum / 2]);
+        return priceInfo.twap;
     }
 
-    function _quickSort(
-        uint256[] memory arr,
-        int256 left,
-        int256 right
-    ) internal pure {
-        int256 i = left;
-        int256 j = right;
-        if (i == j) return;
-        uint256 pivot = arr[uint256(left + (right - left) / 2)];
-        while (i <= j) {
-            while (arr[uint256(i)] < pivot) i++;
-            while (pivot < arr[uint256(j)]) j--;
-            if (i <= j) {
-                (arr[uint256(i)], arr[uint256(j)]) = (
-                    arr[uint256(j)],
-                    arr[uint256(i)]
-                );
-                i++;
-                j--;
-            }
+    /// @notice Allows anyone to initial price to make estimate gas easy on provider chain
+    function initialPrice(address[] calldata _asset) external {
+        uint256 assetLength = _asset.length;
+        for (uint256 index = 0; index < assetLength; index++) {
+            _initialPrice(_asset[index]);
         }
-        if (left < j) _quickSort(arr, left, j);
-        if (i < right) _quickSort(arr, i, right);
     }
 
-    modifier whenNotPaused(address _asset) {
+    function _initialPrice(address _asset) internal {
+        PriceInformation memory priceInfo = assetPriceMap[_asset];
         require(
-            !assetFeederMap[_asset].paused,
-            "NFTOracle: nft price feed paused"
+            priceInfo.updatedTimestamp == 0 && priceInfo.twap == 0,
+            "NFTOracle: asset price is initialized"
         );
-        _;
+        priceInfo.twap == 1;
     }
```

Comment:
what's the purpose of  `initialPrice` function? Just get the gas cost of feeding price?